TwilioVerifySDK/TwilioVerify/Sources/Domain/Factor/FactorMigrations.swift

```diff
@@ -24,3 +24,36 @@ struct FactorMigrations {
     []
   }
 }
+
+class AddKeychainServiceToFactors: Migration {
+  
+  private let secureStorage: SecureStorageProvider
+  private let factorMapper: FactorMapper
+  
+  init(secureStorage: SecureStorageProvider,
+       factorMapper: FactorMapper = FactorMapper()) {
+    self.secureStorage = secureStorage
+    self.factorMapper = factorMapper
+  }
+  
+  var startVersion: Int = 0
+  
+  var endVersion: Int = 1
```

Comment:
Those these two values should reflect the `currentStorage` version of the device and the `currentVersion` of the SDK?

Reply:
They are the startVersion to run the migration and the end version it should apply. Imagine having a new storage version `2` that adds a new field in a factor and another version `3` that removes another field. If you start with the version 1 and install the new app version using storage version `3` you will be missing these two changes. The first migration will move version `1` to `2` and will change the storage version to `2`. The next migration will move version from `2` to `3` removing the field. For now it is the implementation for handling changes in the storage, we can discuss further and we can change/improve it in the future (e.g. supporting mutable-content)